Allow Upper Case property names

insekticid opened the pull request:
ReflectionExtractor::getProperties() returns $id instead of $Id. It is bad naming convention, but is possible

```
class Entity {
    protected $Id;

    public function getId()
    {
        return $this->Id;
    }
}
```

| Q             | A
| ------------- | ---
| Branch?       | 2.8
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | no
| License       | MIT

https://github.com/api-platform/core/issues/1026

dunglas reviewed (approved):
👍

dunglas commented:
Can you cover this with an unit test?

insekticid commented:
I have fixed code style and updated code with master + force push to my patch-1 branch

You need manually rerun build checks:

> If you have write access to the repo: On the build's detail screen, there is a button ↻ with the tooltip "Restart Build".

nicolas-grekas commented:
small rebase needed

nicolas-grekas commented:
should be merged on 2.7, isn't it?

nicolas-grekas commented:
:+1:

insekticid commented:
Symfony 2.7+ So what to do now? It will be automatically merged into Symfony 3 too?

fabpot commented:
@insekticid That's correct. We are merging bug fixes in the lowest branch where the bug exist and then merge it back to all other branches (2.8, 3.2, and master right now). Can you change the target of the PR and rebase it? Thanks.

insekticid committed:
Allow Upper Case property names

ReflectionExtractor::getProperties() returns $id instead of $Id. It is bad naming convention, but is possible

```
class Entity {
    protected $Id;

    public function getId()
    {
        return $this->Id;
    }
}
```

# Conflicts:
#	src/Symfony/Component/PropertyInfo/Tests/Extractors/ReflectionExtractorTest.php
#	src/Symfony/Component/PropertyInfo/Tests/Fixtures/Dummy.php

insekticid commented:
@fabpot PropertyInfo exists since Sf 2.8 so I remade patch-1 for remotes/origin/2.8

fabpot commented:
@insekticid You forgot to change the branch of the PR

insekticid commented:
@fabpot where? in commit message?

fabpot commented:
Thank you @insekticid.

fabpot commented:
Nevermind, I've done the switch when merging. FYI, that's just after the PR title, the branch you asked this PR to be merged in.

fabpot commented:
No need to close it. You can now (that was changed recentyl on Github) update the target on a PR.
